core-header-panel.html

```diff
@@ -67,211 +67,261 @@
+      },
+
+      published: {
+        /**
+         * Controls header and scrolling behavior. Options are
+         * `standard`, `seamed`, `waterfall`, `waterfall-tall`, `scroll` and
+         * `cover`. Default is `standard`.
+         *
+         * `standard`: The header is a step above the panel. The header will consume the
+         * panel at the point of entry, preventing it from passing through to the
+         * opposite side.
+         *
+         * `seamed`: The header is presented as seamed with the panel.
+         *
+         * `waterfall`: Similar to standard mode, but header is initially presented as
+         * seamed with panel, but then separates to form the step.
+         *
+         * `waterfall-tall`: The header is initially taller (`tall` class is added to
+         * the header).  As the user scrolls, the header separates (forming an edge)
+         * while condensing (`tall` class is removed from the header).
+         *
+         * `scroll`: The header keeps its seam with the panel, and is pushed off screen.
+         *
+         * `cover`: The panel covers the whole `core-header-panel` including the
+         * header. This allows user to style the panel in such a way that the panel is
+         * partially covering the header.
+         *
+         *     <style>
+         *       core-header-panel[mode=cover]::shadow #mainContainer {
+         *         left: 80px;
+         *       }
+         *       .content {
+         *         margin: 60px 60px 60px 0;
+         *       }
+         *     </style>
+         *
+         *     <core-header-panel mode="cover">
+         *       <core-toolbar class="tall">
+         *         <core-icon-button icon="menu"></core-icon-button>
+         *       </core-toolbar>
+         *       <div class="content"></div>
+         *     </core-header-panel>
+         *
+         * @attribute mode
+         * @type string
+         * @default ''
+         */
+        mode: String,
+
+        /**
+         * If true, the drop-shadow is always shown no matter what mode is set to.
+         *
+         * @attribute shadow
+         * @type boolean
+         * @default false
+         */
+        shadow: Boolean,
+
+        /**
+         * The class used in waterfall-tall mode.  Change this if the header
+         * accepts a different class for toggling height, e.g. "medium-tall"
+         *
+         * @attribute tallClass
+         * @type string
+         * @default 'tall'
+         */
+        tallClass: String
+      },
+
+      animateDuration: 200,
+
+      configure: function() {
+        return {
+          atTop: true,
+          mode: '',
+          shadow: false,
+          tallClass: 'tall'
+        };
+      },
+
+      calcDropShadowClass: function(atTop, mode, shadow) {
+        return classNames({
+          hidden: !this.shadow &&
+            (atTop && MODE_CONFIGS.shadowMode[this.mode] ||
+              MODE_CONFIGS.noShadow[this.mode])
```

Comment:
Can we break out this conditional logic into something a little more verbose and brain-friendly?